Extract AssetsSidebarTab template and improve UI structure

viva-jinyi opened the pull request:
## Summary
- Extract sidebar template into reusable AssetSidebarTemplate component
- Replace PrimeVue Tabs with TextButton for better visual consistency  
- Improve job detail view header layout with better spacing

## Changes
- Created `AssetSidebarTemplate.vue` as a reusable template component
- Replaced PrimeVue Tabs with TextButton components for tab navigation
- Added i18n translation key for "Back to all assets" button
- Improved spacing and layout in job detail view header
- Maintained all existing functionality while cleaning up template structure

## Test Plan
- [ ] Verify tab switching between Imported and Generated tabs works correctly
- [ ] Test job detail view displays properly with Job ID and execution time
- [ ] Confirm "Back to all assets" button returns to main view
- [ ] Check that all existing media asset features remain functional
- [ ] Verify UI consistency with other sidebar tabs

[screen-capture.webm](https://github.com/user-attachments/assets/4ed192e1-a9f7-4fc1-a41e-f732741dd55d)

viva-jinyi committed:
feat: add job folder view for grouped batch outputs in media assets

  - Add folder view to display all outputs from a single batch job
  - Show output count badge on assets with multiple batch outputs
  - Add job ID display and copy functionality in folder view header
  - Display execution time for batch jobs
  - Implement download functionality for output assets only
  - Add inspect action to asset more menu
  - Extract prompt ID from asset IDs using new UUID utility
  - Add comprehensive tests for UUID extraction utilities

viva-jinyi committed:
refactor: replace custom formatExecutionTime with formatDuration utility

  - Use existing formatDuration util from shared utils instead of custom
  implementation
  - Extract execution time formatting to computed property for cleaner
  template
  - Remove 'any' type assertion in MediaAssetActions, use proper types

viva-jinyi committed:
refactor: Extract AssetsSidebarTab template and improve UI structure

  - Extract sidebar template into reusable AssetSidebarTemplate component
  - Replace PrimeVue Tabs with TextButton for better visual consistency
  - Add i18n key for "Back to all assets" button
  - Improve job detail view header layout with better spacing
  - Maintain existing functionality while cleaning up template structure

viva-jinyi committed:
refactor: Create reusable Tab components and update AssetsSidebarTab UI

  - Add new Tab and TabList components with provide/inject pattern for
  state management
  - Replace TextButton-based tabs with new Tab components in
  AssetsSidebarTab
  - Update AssetSidebarTemplate to use semantic color tokens
  (bg-interface-panel-surface)
  - Improve tab styling with proper hover and focus states

viva-jinyi committed:
feature: tab storybook added

viva-jinyi committed:
chore: icon -> button markup

DrJKL commented on src/utils/uuidUtil.ts:
```diff
@@ -0,0 +1,41 @@
+ * Regular expression for matching UUID v4 format
+ * Format: 8-4-4-4-12 (e.g., 98b0b007-7d78-4e3f-b7a8-0f483b9cf2d3)
+ */
+const UUID_REGEX =
```
https://www.npmjs.com/package/uuid#uuidvalidatestr

DrJKL commented on src/platform/assets/composables/useMediaAssets/useAssetsApi.ts:
```diff
@@ -41,21 +41,29 @@ export function useAssetsApi() {
+            outputCount: task.flatOutputs.filter((o) => o.supportsPreview)
+              .length,
+            allOutputs: task.flatOutputs.filter((o) => o.supportsPreview)
+          }
```
Like the other PR, this makes me think that these two composables are secretly the same thing.

DrJKL commented on src/platform/assets/components/MediaAssetMoreMenu.vue:
```diff
@@ -94,24 +95,30 @@ import { computed, inject } from 'vue'
 
+// Only show Copy Job ID for output assets (not for imported/input assets)
+const showCopyJobId = computed(() => {
+  const assetType = (asset.value as any)?.tags?.[0] || context.value?.type
```
Do you need `as any`?

DrJKL commented on src/platform/assets/components/MediaAssetCard.vue:
```diff
@@ -78,12 +79,15 @@
         <!-- Output count (bottom-right) - show on hover even when playing -->
-        <template v-if="showOutputCount" #bottom-right>
+        <template
+          v-if="showOutputCount && outputCount && outputCount > 1"
```
3 arms is usually when I would consider naming the condition.

viva-jinyi committed:
feature: reapply code

viva-jinyi commented on src/components/sidebar/tabs/AssetsSidebarTab.vue:
```diff
@@ -23,8 +55,15 @@
+            :show-output-count="
+              activeTab === 'output' &&
+              !isInFolderView &&
+              (item.user_metadata?.outputCount as number) > 1
```
- Added `getOutputCount()` helper with proper type checking (`typeof count === 'number'`) - returns 0 for non-numeric or missing values
- Extracted `shouldShowOutputCount()` to separate display logic from template
- Removed unsafe type casting (`as number`) in favor of runtime type validation
- Improved readability by moving complex inline expressions to dedicated functions

This ensures the component handles edge cases gracefully when `outputCount` is undefined, null, or non-numeric.

viva-jinyi commented on src/components/sidebar/tabs/AssetsSidebarTab.vue:
```diff
@@ -137,9 +206,64 @@ const handleAssetSelect = (asset: AssetItem) => {
+const copyJobId = async () => {
+  if (folderPromptId.value) {
+    try {
+      await navigator.clipboard.writeText(folderPromptId.value)
```
✅

viva-jinyi committed:
refactor: Apply code review feedback for media assets and accessibility

- Replace type assertions with type guards and proper runtime checks
- Add OutputAssetMetadata interface for structured metadata typing
- Use uuid package instead of custom regex validation
- Add i18n support for toast messages (Job ID copy)
- Implement WAI-ARIA tab pattern with basic accessibility
- Remove complex keyboard navigation in favor of simple tab key navigation
- Add getAssetType utility function for consistent type handling
- Use test.for for parameterized UUID validation tests
- Simplify Tab components to focus on basic accessibility requirements

viva-jinyi committed:
fix: Improve type safety in asset metadata
  validation

  - Replace any type with proper Record<string,
  unknown> | undefined
  - Fix nodeId type check to support both string
  and number types
  - Remove redundant object type checks in type
  guard
  - Remove unused isValidUuidV4 function from
  uuidUtil
  - Simplify metadata validation logic
